buildfile/src/main/scala/sbt/internal/parser/SbtParser.scala

```diff
@@ -257,24 +235,36 @@ private[sbt] case class SbtParser(path: VirtualFileRef, lines: Seq[String])
       lines: Seq[String]
   ): (Seq[(String, Int)], Seq[(String, LineRange)], Seq[(String, Tree)]) = {
     // import sbt.internal.parser.MissingBracketHandler.findMissingText
-    val indexedLines = lines.toIndexedSeq
-    val content = indexedLines.mkString(END_OF_LINE)
+    val code = lines.toIndexedSeq.mkString(END_OF_LINE)
+    val wrapCode = s"""object SyntheticModule {
+                      |$code
+                      |}""".stripMargin
     val fileName = path.id
-    val (parsedTrees, reporterId, sourceFile) = parse(content, fileName, None)
-    given ctx: Context = compileCtx
+    val reporterId = s"$fileName-${Random.nextInt}"
+    val sourceFile = SourceFile(
+      VirtualFile(reporterId, wrapCode.getBytes(StandardCharsets.UTF_8)),
+      scala.io.Codec.UTF8
+    )
+    given Context = compileCtx.fresh.setSource(sourceFile)
```

Comment:
We need to set the source in the context before we call the parser. Otherwise we need to reconstruct the positions of the trees ourselves.